src/stores/array_formula_highlight.ts

```diff
@@ -1,25 +1,126 @@
+import { clipboardHandlersRegistries } from "../clipboard_handlers";
+import { isInside, recomputeZones } from "../helpers";
+import { getClipboardDataPositions } from "../helpers/clipboard/clipboard_helpers";
 import { Get } from "../store_engine";
-import { Highlight, Zone } from "../types";
+import { _t } from "../translation";
+import { Command, Highlight, Zone } from "../types";
 import { CellErrorType } from "../types/errors";
+import { SelectionEvent } from "../types/event_stream";
 import { HighlightStore } from "./highlight_store";
+import { NotificationStore } from "./notification_store";
 import { SpreadsheetStore } from "./spreadsheet_store";
 
 export class ArrayFormulaHighlight extends SpreadsheetStore {
   protected highlightStore = this.get(HighlightStore);
+  private notification = this.get(NotificationStore);
+  private draggedHighlightZone?: Zone;
 
   constructor(get: Get) {
     super(get);
     this.highlightStore.register(this);
   }
 
+  protected handle(cmd: Command): void {
+    switch (cmd.type) {
+      case "START_CHANGE_HIGHLIGHT":
+        const zone = this.getHighlightZone();
+        if (zone && this.getters.isGridSelectionActive()) {
+          this.model.selection.capture(
+            this,
+            { cell: { col: zone.left, row: zone.top }, zone },
+            {
+              handleEvent: this.handleEvent.bind(this),
+              release: this.cancelChangeHighlight.bind(this),
+            }
+          );
+        }
+        break;
+      case "STOP_CHANGE_HIGHLIGHT": {
```

Comment:
https://www.youtube.com/watch?v=4fndeDfaWCg&t=54s 

I was so happy to get rid of that command a year ago :'(

Reply:
Wdym ? The command was still in the codebase before this PR :sweat_smile:

Reply:
Nvm I'm stupid and looked at the wrong command. 

But yeah I'm not really happy about how the highlight d&d work.

Reply:
There is fundamental issue as we want to have a differed behaviour while using a direct channel.
So we add commands to add some differed component

Might not be better but at least no commands - Highlights can be defined as movable and resizable.
What about registering them with callbacks as well? 
```typescript
    const highlight = {
        ...,
        onMove: Object|undefined {start: () => {
            // do internal shit to the concerned store/plugin
        }, stop: () => {
            // finalize stuff inside the store/plugin
        }},
        onResize: Object|undefined {start: () => {
            // do internal shit to the concerned store/plugin
        }, stop: () => {
            // finalize stuff inside the store/plugin
        }},
```

This way, the store can implement it's prefered behaviour without the global command that starts with "if i'm the one listening to the selection, then..." 

I think it can work since highlight are only store related now?

WDYT? cc @LucasLefevre @pro-odoo

Reply:
I also don't like the additional command and temporary state associated. I'm also not sure the feature cost/benefit ratio is worth it to be honest

Reply:
TBH we probably can cancel the task, without the dashboard reorganize feature, drag & drop array formula isn't that useful. We'll still need something similar to drag & drop the selection tho

Reply:
>  We'll still need something similar to drag & drop the selection tho

Why? It seems like the exact same stuff ?

Reply:
> I also don't like the additional command and temporary state associated. I'm also not sure the feature cost/benefit ratio is worth it to be honest

https://media1.tenor.com/m/TY-5qp551-AAAAAC/ron-ronswanson.gif